src/toil/jobStores/azureJobStore.py

```diff
@@ -73,61 +75,215 @@ def _fetchAzureAccountKey(accountName):
+            jobStore.keyPath = jobStore.config.cseKey
+        return jobStore
+
+    # Do not invoke the constructor, use the factory method above.
+
+    def __init__(self, accountName, accountKey, namePrefix, tableService, blobService, jobItems, jobFileIDs, files,
+                 statsFiles, statsFileIDs, config=None, jobChunkSize=maxAzureTablePropertySize):
+        """
+        Creates a new AzureJobStore instance with the given components.
+
+        :param str accountName: Name of Azure account.
+        :param str accountKey: The key that corresponds with the given account name.
+        :param str namePrefix: Job store components will be prefixed with this.
+        :param azure.storage.table.tableservice.TableService tableService: Table service that corresponds
+            with the given account.
+        :param azure.storage.blob.blobservice.BlobService blobService: Blob service that corresponds with
+            the given account.
+        :param azure.storage.table.Table jobItems: Table of serialized jobs.
+        :param azure.storage.table.Table jobFileIDs: Table that maps files to jobs.
+        :param azure.storage.blob.Container files: Container for all shared and unshared files.
+        :param azure.storage.blob.Container statsFiles: Container for stats and logging strings.
+        :param azure.storage.table.Table statsFileIDs: Table of file IDs that contain stats and logging
+            strings.
+        :param toil.common.Config config: the config object to written to this job store.
+            Must be None for existing job stores. Must not be None for new job stores.
+        :param int jobChunkSize: Size of a chunk for splitting up the serialized job into chunks
+            that each fit into a property value of the an Azure table entity.
+        """
+
+        self.jobChunkSize = jobChunkSize
+        self.keyPath = None
+        self.account_key = accountKey
+        self.accountName = accountName
+        self.namePrefix = namePrefix
+        logger.debug("Creating job store with name prefix '%s'" % self.namePrefix)
+
+        # These are the main API entrypoints.
+        self.tableService = tableService
+        self.blobService = blobService
+
+        # containers
+        self.files = files
+        self.statsFiles = statsFiles
+
+        # tables
+        self.jobItems = jobItems
+        self.jobFileIDs = jobFileIDs
+        self.statsFileIDs = statsFileIDs
+
+        self.updateRegistry(self.namePrefix, self.tableService, exists=True)
+        super(AzureJobStore, self).__init__()
+
+    @classmethod
+    def jobStoreExists(cls, locator):
+        info = cls._processLocator(locator)
+        registryTable = cls._getOrCreateTable(info.tableService, 'toilRegistry')
+        for attempt in retry_on_error():
+            with attempt:
+                return registryTable.get_entity(row_key=info.namePrefix) is not None
+
+    @classmethod
+    def updateRegistry(cls, namePrefix, tableService, exists):
```

Comment:
This PR does not add a registry to any job store that did not already have one. The ```updateRegistry``` method merely replaces code the ```__init__``` and ```deleteJobStore``` methods of the AWS and Azure job stores which both already maintain registries. Its main advantage is that it allows the registry to be updated without instantiating a job store since it is a class method.